src/pools/D3MCompoundPool.sol

```diff
@@ -0,0 +1,200 @@
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPool.sol";
+
+interface TokenLike {
+    function balanceOf(address)         external view returns (uint256);
+    function approve(address, uint256)  external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+}
+
+interface VatLike {
+    function live() external view returns (uint256);
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface D3mHubLike {
+    function vat() external view returns (address);
+}
+// cDai - https://etherscan.io/token/0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643
+interface CErc20Like is TokenLike {
+    function underlying()                           external view returns (address);
+    function comptroller()                          external view returns (address);
+    function exchangeRateStored()                   external view returns (uint256);
+    function getCash()                              external view returns (uint256);
+    function getAccountSnapshot(address account)    external view returns (uint256, uint256, uint256, uint256);
+    function mint(uint256 mintAmount)               external returns (uint256);
+    function redeemUnderlying(uint256 redeemAmount) external returns (uint256);
+    function accrueInterest()                       external returns (uint256);
+    function exchangeRateCurrent()                  external returns (uint256);
+}
+
+// Comptroller - https://etherscan.io/address/0x3d9819210a31b4961b30ef54be2aed79b9c9cd3b
+interface ComptrollerLike {
+    function getCompAddress() external view returns (address);
+    function claimComp(address[] memory holders, address[] memory cTokens, bool borrowers, bool suppliers) external;
+}
+
+contract D3MCompoundPool is ID3MPool {
+
+    mapping (address => uint256) public wards;
+    address                      public hub;
+    address                      public king; // Who gets the rewards
+
+    VatLike         public immutable vat;
+    ComptrollerLike public immutable comptroller;
+    TokenLike       public immutable comp;
+    TokenLike       public immutable dai;
+    CErc20Like      public immutable cDai;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event Collect(address indexed king, address indexed gift, uint256 amt);
+
+    constructor(address hub_, address cDai_) {
+        cDai        = CErc20Like(cDai_);
+        dai         = TokenLike(cDai.underlying());
+        comptroller = ComptrollerLike(cDai.comptroller());
+        comp        = TokenLike(comptroller.getCompAddress());
+
+        require(address(comp) != address(0), "D3MCompoundPool/invalid-comp");
+
+        dai.approve(cDai_, type(uint256).max);
+
+        hub = hub_;
+        vat = VatLike(D3mHubLike(hub).vat());
+        vat.hope(hub);
```

Comment:
nit
```suggestion
        vat = VatLike(D3mHubLike(hub_).vat());
        vat.hope(hub_);
```
It is more gas efficient, and same way than done in aave.

Reply:
Gas seems to be the same, changed it for alignment with aave in `8ce3449`.

Reply:
Ah yes? so there isn't a SLOAD in the constructor. Good to know.